src/sdk-reference/cpp/1/essentials/error-handling/index.md

```diff
@@ -0,0 +1,39 @@
+---
+layout: sdk.html.hbs
+algolia: true
+title: Error Handling
+description: How to handle errors with the SDK
+order: 100
+---
+
+# Error Handling
+
+Each of the SDK methods is likely to fail and throw an exception.  
```

Comment:
I hope that our SDK methods aren't likely to fail! ;-)

Also:

* since exceptions are the standard way of handling errors in C++, we shouldn't have to mention that
* since C++ is strongly-typed, we have to provide the properties type
* I don't think providing the complete definition of the abstract `KuzzleResponse` class is useful: this chapter is about handling exceptions, not overloading or instantiating ones.
* we should inherit from `std::exception`, not from `std::runtime_error`
* we should overload the `what` method, instead of exposing a new, non-standard, `getMessage` one 

Proposal, assuming that the definition of the `KuzzleException` class is fixed:

```
SDK methods handle failure by throwing exceptions inheriting the `KuzzleException` class, which implements the standard `std::exception` interface.

Members:

* `unsigned int status`: error code, following [HTTP Standards](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)
* `const char * what() const;`: returns the error message
```
All exceptions thrown by our SDKs feature the following properties:
```